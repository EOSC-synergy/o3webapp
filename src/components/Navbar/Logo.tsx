import React, { FC } from 'react';
import PropTypes from 'prop-types';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

export const LOGO_SOURCE =
    process.env.NEXT_PUBLIC_LOGO_SRC ?? 'https://o3as.data.kit.edu/img/logos/o3as-logo.svg';
export const LOGO_NAVIGATION = '#page-top';
const LOGO_TEXT = 'O3as: Webapp';

type LogoProps = {
    // TODO: Object typing? see https://mui.com/system/display/
    display?: string | object;
};
/**
 * Returns a logo from O3AS and the text "O3AS Webapp".
 *
 * @param display See https://mui.com/system/display/
 */
const Logo: FC<LogoProps> = ({ display }) => {
    return (
        <Link
            href={LOGO_NAVIGATION}
            sx={{
                flexGrow: 1,
                textDecoration: 'inherit',
                color: 'inherit',
                display: display,
            }}
            noWrap
            data-testid="logo-container"
        >
            <Grid container sx={{ alignItems: 'center', margin: '0.5em', marginLeft: '2em' }}>
                <img
                    data-testid="logo-image"
                    style={{ height: '4em', margin: '0.5em' }}
                    src={LOGO_SOURCE}
                    alt="logo"
                />
                <Typography
                    variant="h6"
                    noWrap
                    data-testid="logo-text"
                    sx={{
                        color: '#C7D3E0',
                        '&:hover': {
                            color: '#fed136',
                        },
                        fontSize: '28px',
                    }}
                >
                    {LOGO_TEXT}
                </Typography>
            </Grid>
        </Link>
    );
};

Logo.propTypes = {
    /** Display style to pass to the logo */
    display: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};

export default Logo;
