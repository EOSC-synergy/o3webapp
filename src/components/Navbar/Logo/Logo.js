import React from 'react';
import PropTypes from 'prop-types';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

/**
 * The link that provides the image logo
 * @constant {string}
 * @memberof Logo
 */
export const logoSrc = "https://o3as.data.kit.edu/img/logos/o3as-logo.svg";
/**
 * The link to the O3AS Webapp
 * @constant {string}
 * @memberof Logo
 */
export const O3ASLink = "#page-top";

/**
 * Returns a logo from O3AS and the text "O3AS Webapp"
 * @component
 * @param {Object} props 
 * @param {Object|String} props.display when the logo should be displayed
 * @returns {JSX.Element} containing one Image and one Text wrapped in a Link component
 */
const Logo = (props) => {
    return (
        <Link
            href={O3ASLink}
            sx={{ flexGrow: 1, textDecoration: 'inherit', color: 'inherit', display: props.display}}
            noWrap
            data-testid="logo-container"
        >   
            <Grid container sx={{ alignItems: "center", margin: '0.5em', marginLeft: '2em'}}>
                <img data-testid="logo-image" style={{height: '4em', margin: '0.5em'}} src={logoSrc} alt={"logo"} />
                <Typography
                    variant="h6"
                    noWrap
                    data-testid="logo-text"
                    sx={{
                        color: '#C7D3E0',
                        '&:hover': {
                            color: '#fed136',
                        },
                        fontSize: '28px'
                    }}
                >
                    O3as: Webapp
                </Typography>
            </Grid>
        </Link>
    );
};

Logo.propTypes = {
    display: PropTypes.oneOfType([
        PropTypes.object,
        PropTypes.string
    ])
}

export default Logo;