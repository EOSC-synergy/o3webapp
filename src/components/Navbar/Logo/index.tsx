import React from 'react';
import PropTypes from 'prop-types';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

/**
 * The link that provides the image logo
 * @constant {String}
 * @memberof Logo
 * @default "https://o3as.data.kit.edu/img/logos/o3as-logo.svg"
 */
export const logoSrc = 'https://o3as.data.kit.edu/img/logos/o3as-logo.svg';

/**
 * The link to the O3AS Webapp
 * @constant {String}
 * @memberof Logo
 * @default "#page-top"
 */
export const O3ASLink = '#page-top';

type LogoProps = {
    // TODO: Object typing? see https://mui.com/system/display/
    display?: string | Object;
};
/**
 * Returns a logo from O3AS and the text "O3AS Webapp".
 * @component
 * @param display see https://mui.com/system/display/
 * @returns {JSX.Element} JSX Element containing one image and one text wrapped in a Link component
 */
const Logo: React.FC<LogoProps> = ({ display }): JSX.Element => {
    /**
     * Name of the webapp that should be displayed in the logo
     * @constant {String}
     * @default "O3as: Webapp"
     */
    const logoName = 'O3as: Webapp';

    return (
        <Link
            href={O3ASLink}
            sx={{
                flexGrow: 1,
                textDecoration: 'inherit',
                color: 'inherit',
                display: display,
            }}
            noWrap
            data-testid="logo-container"
        >
            <Grid container sx={{ alignItems: 'center', margin: '0.5em', marginLeft: '2em' }}>
                <img
                    data-testid="logo-image"
                    style={{ height: '4em', margin: '0.5em' }}
                    src={logoSrc}
                    alt={'logo'}
                />
                <Typography
                    variant="h6"
                    noWrap
                    data-testid="logo-text"
                    sx={{
                        color: '#C7D3E0',
                        '&:hover': {
                            color: '#fed136',
                        },
                        fontSize: '28px',
                    }}
                >
                    {logoName}
                </Typography>
            </Grid>
        </Link>
    );
};

Logo.propTypes = {
    /**
     * display style to pass to the logo
     */
    display: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};

export default Logo;
