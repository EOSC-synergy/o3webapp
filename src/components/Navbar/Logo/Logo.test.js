import React from 'react';
import ReactDOM from 'react-dom';
import Logo from './Logo';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { logoSrc, O3ASLink } from './Logo';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Logo />, div);
});

// Snapshot test
it('renders correctly', () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
});


it('displays the name of O3AS Webapp', () => {
    render(<Logo />);
    expect(screen.getByTestId("logo-text")).toHaveTextContent("O3as: Webapp");
});

it('displays the correct image', () => {
    render(<Logo />);
    const image = screen.getByAltText('logo');
    expect(image.src).toContain(logoSrc);
});

it('renders correct link onto logo', () => {
    render(<Logo />);
    expect(screen.getByText("O3as: Webapp").closest('a')).toHaveAttribute('href', O3ASLink);
});

it('renders nothing if props.display={"none"}', () => {
    render(<Logo display={'none'} />);
    expect(screen.getByTestId("logo-container")).not.toBeVisible();
})