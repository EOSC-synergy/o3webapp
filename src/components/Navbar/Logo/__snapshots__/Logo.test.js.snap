// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiTypography-noWrap MuiLink-root MuiLink-underlineAlways css-q2mzyx-MuiTypography-root-MuiLink-root"
    data-testid="logo-container"
    href="#page-top"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1e0cb12-MuiGrid-root"
    >
      <img
        alt="logo"
        data-testid="logo-image"
        src="https://o3as.data.kit.edu/img/logos/o3as-logo.svg"
        style="height: 4em; margin: 0.5em;"
      />
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-154cv3q-MuiTypography-root"
        data-testid="logo-text"
      >
        O3as: Webapp
      </h6>
    </div>
  </a>
</DocumentFragment>
`;
